src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,230 @@
+            saveCarStatement.setLong(1, car.getManufacturer().getId());
+            saveCarStatement.setString(2, car.getModel());
+            saveCarStatement.executeUpdate();
+            ResultSet resultSet = saveCarStatement.getGeneratedKeys();
+            if (resultSet.next()) {
+                car.setId(resultSet.getObject(1, Long.class));
+            }
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't create "
+                    + car + ". ", throwable);
+        }
+        return car;
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String query = "SELECT cars.id AS id, "
+                + "manufacturers.`name` AS manufacturer_name,  "
+                + "cars.model AS model,  "
+                + "manufacturers.country AS country, "
+                + "manufacturers.id AS manufacturers_id "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.id = ? AND cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement = connection.prepareStatement(query)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            Car car = new Car();
+            if (resultSet.next()) {
+                car = parseCarFromResultSet(resultSet);
+            }
+            return Optional.of(car);
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get car by ID: " + id, throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT cars.id AS id, "
+                + "manufacturers.`name` AS manufacturer_name, "
+                + "cars.model AS model, "
+                + "manufacturers.country AS country, "
+                + "manufacturers.id AS manufacturers_id "
+                + "FROM cars "
+                + "JOIN manufacturers "
+                + "ON cars.manufacturer_id = manufacturers.id "
+                + "WHERE cars.is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            List<Car> cars = new ArrayList<>();
+            while (resultSet.next()) {
+                cars.add(parseCarFromResultSet(resultSet));
+            }
+            return cars;
+        } catch (SQLException throwable) {
+            throw new DataProcessingException("Couldn't get a list of cars from cars DB.",
+                    throwable);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        List<Car> cars = getAll();
```

Comment:
Imagine that you have 1,000,000 cars in your database, and every time a user in your service wants to look at a list of his cars, your app get`s all cars from the database and then filter them...

Reply:
Fixed.